### PR TITLE
Throw barrels animations

### PR DESCRIPTION
**Pull request by JoanMarcBardes**

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-1g0119g Throw barrels animations](https://app.clickup.com/t/1g0119g)

**Comment by jcarlos0305** [comment]
Task linked: [CU-p1btku Boss mechanics](https://app.clickup.com/t/p1btku)

**Review by Mefiso — CHANGES_REQUESTED** [review]
A couple of things: 
1. I know it isn't the final effect but please make that the warning effect ("barrel shadow") lasts exactly until the barrel explodes or reaches the floor, as you prefer, not more. Do it by scripting (set the warning effect duration equal to the timerToDestroy or the falling time).
2. In the condition to check if the barrel has reached the floor, use the parentTransform barrelMesh's transform (changing the threshold accodingly). In the "else" statement force the global position of the parentTransform to be exactly that of the new threshold.

File: Penteract/Assets/Scripts/Barrel.cpp [hunk 1]
```diff
@@ -49,7 +50,18 @@ void Barrel::Start() {
 		audioForTimer = particleForTimerAux->GetComponent<ComponentAudioSource>();
 	}
 
+	if (!onFloor) {
+		float3 pos = parentTransform->GetGlobalPosition();
+		parentTransform->SetGlobalPosition(float3(pos.x, heightOfThrow, pos.z));
+	}
 
+	barrelShadow = barrel->GetParent()->GetParent()->GetChild("BarrelShadowWarning");
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/Barrel.cpp:58]: Why is this not inside "if(!onFloor)"

File: Penteract/Assets/Scripts/Barrel.cpp [hunk 2]
```diff
@@ -90,12 +102,17 @@ void Barrel::Update() {
...
+			}
 		}
 	}
 
 	if (!onFloor) {
-		if (barrelMesh->GetComponent<ComponentTransform>()->GetGlobalPosition().y > 0.25f) {
+		if (parentTransform->GetGlobalPosition().y > 2.0f) {
 			float3 barrelPos = parentTransform->GetGlobalPosition();
 			barrelPos += float3(0, -forceOfFall, 0);
 			parentTransform->SetGlobalPosition(barrelPos);
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/Barrel.cpp:118]: PLEASE, on the else enforce that the y component of the global position is EXACTLY the threshold (2.f). If not, in some computer the black mark won't be seen on the floor

**Review by Mefiso — APPROVED** [review]

**Review by isaacfdez — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by devildrake — APPROVED** [review]

**Pull request merged**